tests/unit/aiplatform/test_jobs.py

```diff
@@ -988,48 +1041,22 @@ def get_mdm_job_mock():
     with mock.patch.object(
         _TEST_API_CLIENT, "get_model_deployment_monitoring_job"
     ) as get_mdm_job_mock:
-        get_mdm_job_mock.return_value = (
-            gca_model_deployment_monitoring_job_compat.ModelDeploymentMonitoringJob(
-                name=_TEST_MDM_JOB_NAME,
-                display_name=_TEST_DISPLAY_NAME,
-                state=_TEST_JOB_STATE_RUNNING,
-                endpoint=_TEST_ENDPOINT,
-            )
-        )
+        get_mdm_job_mock.side_effect = [
+            _TEST_MDM_OLD_JOB,
+            _TEST_MDM_OLD_JOB,
+            _TEST_MDM_OLD_JOB,
+            _TEST_MDM_OLD_JOB,
+            _TEST_MDM_EXPECTED_NEW_JOB,
+        ]
         yield get_mdm_job_mock
 
 
 @pytest.fixture
-@pytest.mark.usefixtures("get_mdm_job_mock")
 def update_mdm_job_mock(get_endpoint_with_models_mock):  # noqa: F811
     with mock.patch.object(
         _TEST_API_CLIENT, "update_model_deployment_monitoring_job"
     ) as update_mdm_job_mock:
-        expected_objective_config = gca_model_monitoring_compat.ModelMonitoringObjectiveConfig(
-            prediction_drift_detection_config=gca_model_monitoring_compat.ModelMonitoringObjectiveConfig.PredictionDriftDetectionConfig(
-                drift_thresholds={
-                    "TEST_KEY": gca_model_monitoring_compat.ThresholdConfig(value=0.01)
-                }
-            )
-        )
-        all_configs = []
-        for model in get_endpoint_with_models_mock.return_value.deployed_models:
-            all_configs.append(
-                gca_model_deployment_monitoring_job_compat.ModelDeploymentMonitoringObjectiveConfig(
-                    deployed_model_id=model.id,
-                    objective_config=expected_objective_config,
-                )
-            )
-
-        update_mdm_job_mock.return_vaue.result_type = (
-            gca_model_deployment_monitoring_job_compat.ModelDeploymentMonitoringJob(
-                name=_TEST_MDM_JOB_NAME,
-                display_name=_TEST_DISPLAY_NAME,
-                state=_TEST_JOB_STATE_RUNNING,
-                endpoint=_TEST_ENDPOINT,
-                model_deployment_monitoring_objective_configs=all_configs,
-            )
-        )
+        update_mdm_job_mock.return_value.result_type = _TEST_MDM_EXPECTED_NEW_JOB
```

Comment:
`result_type` is intended to be a type not an instance of that type. Refer:

https://github.com/googleapis/python-api-core/blob/main/google/api_core/operation.py#L60

Here is an example of mocking an lro: https://github.com/googleapis/python-aiplatform/blob/main/tests/unit/aiplatform/test_datasets.py#L370

Reply:
done